### ISSUE TITLE
Operation Insights: Metadata Tags for operations

### ISSUE THREAD

**Issue by erikwrede** [kamilkisiela/graphql-hive#4931]
It would be great if Hive reports were taggable and filterable by custom user-defined tags. Use cases for tags could include

- Region (e.g. europe-west, asia-pacific, north-america) for examining perfomance per Region
- Backend-Revision - examining revision-specific performance 
- Tenant-name - enabling multitenancy analysis

Essentially, tags complement `ClientInfo` to provide additional metadata on the request. An operation could have one or multiple tags, resulting in the following changes to the Hive Reporting Spec:

```diff
export interface RequestOperation {
  timestamp: number
  operationMapKey: string
  execution: Execution
  metadata?: Metadata
}
 
export interface Metadata {
  client?: Client
+ tag?: string // OR
+ tags?: string[]
}
```

In the first iteration, the insights UI could be adjusted to have an additional filter like this:

<img width="990" alt="image" src="https://github.com/kamilkisiela/graphql-hive/assets/9725143/cf4e6bc1-f35c-4883-b375-e040520abfc8">

If you agree on this feature being useful, there's several points to discuss:

- One vs Multiple tags - (utility vs DB index performance)
- Necessity of additional ClickHouse materialized views for Tags (complementing client, operation views)
- Default tag if none was provided (better DB performance), similar to client -> "unknown"

Looking forward to your opinion on this. 

Happy to draft an initial PR once we are aligned on the open points.

**Comment by kamilkisiela** [comment]
Makes sense, it shouldn't be an issue to maintain it and it's relatively easy to implement.
Yeah, we could accept such PR.

- Multiple tags (it's better future-wise)
- index for tags in materialized views of ClickHouse

and we're good to go

**Comment by erikwrede** [comment]
Small update, I still have this on my list and have started work. Some other things sadly got in between, but I'm still hoping to push this through soon!

**Comment by n1ru4l** [comment]
We should also think about whether we want to have just tags or maybe even labels with a key and value.

Some examples:

| key | value |
|----|----|
|  `customer`| actual identifier of that customer |
| `protocol` |  `websocket`, `http`, ... |

That way, we could provide a nicer UI for the user where you can select filters based on a category (key) instead of a list of tags. Behind the scenes the search would still act as based on tags though.

```
export interface Metadata {
  client?: Client
  tags?: {
    [key: string]: Array<string> | string;
  }
}
```

In clickhouse we could store them in an array column with the operation data (for filtering) like this:

```
`${key}|${value}`
```

But also store the keys and categories separately for easy lookup on the dashboard UI.

```
target_operation_labels(target_id, label_key, label_value)
```

**Comment by kamilkisiela** [comment]
We should do what stellate has with custom attributes
